### PR TITLE
HDDS-9726. Add page jump function and entity count display to the Node Status section of the SCM UI.

### PR DESCRIPTION
**Pull request by aierate**
## What changes were proposed in this pull request?
If there are hundreds of Datanodes, it is difficult to browse the situation and view the number of Datanodes.
In the SCM UI, I need to check how many datanodes there are, so here I need to display the number of records of the datanodes and be able to jump to the specified page during pagination.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9726

## How was this patch tested?
Only all unit test.
https://github.com/aierate/ozone/actions/runs/6938742317

Here is UI screenshot.

image

### CODE REVIEW

**Comment by adoroszlai** [comment]
@devabhishekpal please take a look

**Comment by devabhishekpal** [comment]
Hello @aierate.

Some things:
- One thing I notice is that the default value for the pagination is blank as attached in your screenshot.
  Can we fix that to have a default value of 1?
- Also I tested this out locally, it seems like clicking the pagination works - as in the page changes, however in the dropdown it doesn't update the page number. So it stays blank irrespective of which page it jumped to.
    - This might happen because currentPage on `this` scope is becoming `undefined` it seems. And since undefined is not a part of any of the returned values by `getPagesArray()` which is being used in the <options> it is being set as blank. Could you please look into this and fix that?

Thanks a lot for working on this patch.

**Comment by aierate** [comment]
> Hello @aierate.
> 
> Some things:
> 
> * One thing I notice is that the default value for the pagination is blank as attached in your screenshot.
>   Can we fix that to have a default value of 1?
> * Also I tested this out locally, it seems like clicking the pagination works - as in the page changes, however in the dropdown it doesn't update the page number. So it stays blank irrespective of which page it jumped to.
>   
>   * This might happen because currentPage on `this` scope is becoming `undefined` it seems. And since undefined is not a part of any of the returned values by `getPagesArray()` which is being used in the  it is being set as blank. Could you please look into this and fix that?
> 
> Thanks a lot for working on this patch.

Thank you very much @devabhishekpal for pointing out my bug. I solved this problem by adding (ng value="page") at rows 70 in scm-overview.html . Please help me review again, thank you.

**Comment by aierate** [comment]
> > Hello @aierate.
> > Some things:
> > 
> > * One thing I notice is that the default value for the pagination is blank as attached in your screenshot.
> >   Can we fix that to have a default value of 1?
> > * Also I tested this out locally, it seems like clicking the pagination works - as in the page changes, however in the dropdown it doesn't update the page number. So it stays blank irrespective of which page it jumped to.
> >   
> >   * This might happen because currentPage on `this` scope is becoming `undefined` it seems. And since undefined is not a part of any of the returned values by `getPagesArray()` which is being used in the  it is being set as blank. Could you please look into this and fix that?
> > 
> > Thanks a lot for working on this patch.
> 
> Thank you very much @devabhishekpal for pointing out my bug. I solved this problem by adding (ng value="page") at rows 70 in scm-overview.html . Please help me review again, thank you.

Here is UI screenshot.  @devabhishekpal 

image

**Review by devabhishekpal — APPROVED** [review]
Thanks for working on this @aierate.
Looks good to me now. +1

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html [hunk 1]
```diff
@@ -62,22 +62,35 @@ <h2>Node Status</h2>
...
+
+<div class="row">
+    <div class="col-md-6 text-left">
+        <label>Page:</label>
+        <select class="form-select" ng-model="currentPage" ng-change="handlePagination(currentPage, false)">
+            <option ng-repeat="page in getPagesArray()" ng-value="page">{{page}}</option>
+        </select>
+        <span>of {{lastIndex}}. </span>
+        <span ng-if="nodeStatus && nodeStatus.length > 0">
+            Showing {{getCurrentPageFirstItemIndex()}} to {{getCurrentPageLastItemIndex()}} of {{totalItems}} entries.
```
Comment (dombizita) [inline hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html:74]: I'll leave it up to you if you want to change this, when there are just a few datanodes or showing all the of them in one page (eg. `Showing 1 to 3 of 3 entries.`) I think this is a bit easier to understand. 
```suggestion
            Showing {{getCurrentPageFirstItemIndex()}} to {{getCurrentPageLastItemIndex()}} of the total {{totalItems}} entries.
```

Or we could do this:
```suggestion
            {{getCurrentPageFirstItemIndex()}} - {{getCurrentPageLastItemIndex()}} of {{totalItems}} entries.
```

**Review by dombizita — COMMENTED** [review]
Thanks for working on this @aierate, overall it looks good to me, I also tried it and it is working as expected. I added one small comment.

File: hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html (hunk 1 above)
Comment (aierate) [reply hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm-overview.html:74]: Thanks @dombizita. I think this is enough, I referred to HDFS UI.
image

**Comment by aierate** [comment]
> Thanks for working on this @aierate, overall it looks good to me, I also tried it and it is working as expected. I added one small comment.

Thanks @dombizita. I have thought it over and found your suggestion to be correct, thank you very much. Can you help me merge this pull request?

**Review by dombizita — APPROVED** [review]
Thanks for working on this @aierate, it looks good to me!

**Comment by dombizita** [comment]
Thanks for adding this improvement @aierate! Thanks for the review @devabhishekpal!

**Pull request merged**